scripts/toolchain/install-toolchain.sh

```diff
@@ -0,0 +1,174 @@
+#!/bin/bash
+
+# prevent sourcing of this script, only allow execution
+$(return >/dev/null 2>&1)
+test "$?" -eq "0" && { echo "This script should only be executed." >&2; exit 1; }
+
+# exit on errors, undefined variables, ensure errors in pipes are not hidden
+set -Eeuo pipefail
+
+# set log id and use shared log function for readable logs
+declare mydir
+mydir=$(cd "$(dirname "${BASH_SOURCE[0]}")" &>/dev/null && pwd -P)
+
+# Installs all toolchain utilities that are required to build hoprnet monorepo, including
+# - Node.js
+# - Yarn
+# - Typescript + related utilities, such as ts-node
+# - Rust (rustc, cargo)
+# - wasm-pack + wasm-opt, necessary to build WebAssembly modules
+
```

Comment:
Which OS is supported? Which binaries are going to be located where?

Reply:
changed